src/AudioUnit/AudioComponent.cs

```diff
@@ -224,6 +379,58 @@ public XamCore.AppKit.NSImage GetIcon ()
 			return new XamCore.AppKit.NSImage (AudioComponentGetIcon (handle));
 		}
 #endif
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		[DllImport (Constants.AudioUnitLibrary, EntryPoint = "AudioUnitExtensionSetComponentList")]
+		static extern int /* OSStatus */ SetComponentList (IntPtr /* CFString */ extensionIdentifier, /* CFArrayRef */ IntPtr audioComponentInfo);
+
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		[DllImport (Constants.AudioUnitLibrary, EntryPoint = "AudioUnitExtensionCopyComponentList")]
+		static extern /* CFArrayRef */ IntPtr CopyComponentList (IntPtr /* CFString */ extensionIdentifier);
+
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		public AudioComponentInfo[] ComponentList {
+			get {
+				using (var cfString = new CFString (Name)) {
+					var cHandle = CopyComponentList (cfString.Handle);
+					if (cHandle == IntPtr.Zero)
+						return null;
+					using (var nsArray = Runtime.GetNSObject<NSArray> (cHandle, owns: true)) {
+						if (nsArray == null)
+							return null;
+						// make things easier for developers since we do not know how to have an implicit conversion from NSObject to AudioComponentInfo
+						var dics = NSArray.FromArray <NSDictionary> (nsArray);
+						var result = new AudioComponentInfo [dics.Length];
+						for (var i = 0; i < result.Length; i++) {
+							result [i] = new AudioComponentInfo (dics[i]);
+						}
+						return result;
+					}
+				}
+			}
+			set {
+				using (var cfString = new CFString (Name)) {
+					var dics = new NSDictionary [value.Length];
+					for (var i = 0; i < value.Length; i++) {
+						dics [i] = value [i].Dictionary;
+					}
+					using (var array = NSArray.FromNSObjects (dics)) {
+						var result = (AudioConverterError) SetComponentList (cfString.Handle, array.Handle);
+						switch (result) {
+						case AudioConverterError.HardwareInUse:
+							throw new InvalidOperationException ("ComponentList could not be set: 'Hardware is in use.'");
+						case AudioConverterError.NoHardwarePermission:
+							throw new InvalidOperationException ("ComponentList could not be set: 'No Hardware permission.'");
+						case AudioConverterError.None:
+							return;
+						default:
+							throw new InvalidOperationException ($"ComponentList could not be set, error {result.ToString ()}");
```

Comment:
not sure there's much value in all the similar (but different) strings, they just end up taking 2xspace in the metadata (unicode).
`throw new InvalidOperationException (result.ToString ());` 
should give the required information back to the developer (the stack trace will point to `ComponentList`).